gpii/node_modules/gsettingsBridge/nodegsettings/nodegsettings.cc

```diff
@@ -15,119 +16,131 @@
  */
 
 
-#include <node.h>
-#include <v8.h>
-
+#include <nan.h>
 #include <gio/gio.h>
 
 using namespace v8;
+using v8::FunctionTemplate;
+using v8::Handle;
+using v8::Isolate;
+using v8::Object;
+using v8::String;
+using Nan::GetFunction;
+using Nan::New;
+using Nan::Set;
+using Nan::ThrowError;
 
-Handle<Value> get_gsetting_keys(const Arguments& args) {
-  HandleScope scope;
+NAN_METHOD(get_gsetting_keys) {
+  Isolate *isolate = info.GetIsolate();
   GSettings *settings;
-  char schema[1024];
   gchar ** keys;
   gint i;
   gint size = 0;
-  args[0]->ToString()->WriteAscii(schema);
-  settings = g_settings_new(schema);
+  Local<String> schema = info[0].As<String>();
+  settings = g_settings_new(*String::Utf8Value(schema));
   keys = g_settings_list_keys(settings);
+
   for (i = 0; keys[i]; i++) {
     size++; // Figure out how to do this in 1 loop
   }
-  Handle<Array> togo;
-  togo = Array::New(size);
+  Handle<Array> togo = Array::New(isolate, size);
   for (i = 0; keys[i]; i++) {
-    togo->Set(i,String::New(keys[i]));
+    togo->Set(i,String::NewFromUtf8(isolate, keys[i]));
```

Comment:
I don't see this strategy for constructing strings advertised in the public NAN docs - where did you find the recommendation?

What I see in the docs is this: https://github.com/nodejs/nan/blob/master/doc/string_bytes.md#api_nan_encode which seems to avoid the need for the bizarre "Isolate" reference...